tests/test_structures.py

```diff
@@ -41,20 +45,39 @@ def test_structure_manager_widget(structure_data_object):
         structure_manager_widget.structure[0].position == new_structure[0].position
     )
 
+    # Store the modified structure.
+    structure_manager_widget.btn_store.click()
+    assert structure_manager_widget.structure_node.is_stored
+    stored_structure = structure_manager_widget.structure_node
+
     # Undo the structure modification.
     structure_manager_widget.undo()
     assert np.any(
         structure_manager_widget.structure[0].position != new_structure[0].position
     )
+    structure_manager_widget.undo()  # Undo the structure creation.
+    assert structure_manager_widget.structure is None
 
-    # test the widget can be instantiated with empty inputs
+    # Test the widget with multiple importers, editors. Specify the viewer and the node class
+    base_editor = awb.BasicStructureEditor(title="Basic Editor")
     structure_manager_widget = awb.StructureManagerWidget(
         importers=[
             awb.StructureUploadWidget(title="From computer"),
+            awb.StructureBrowserWidget(title="AiiDA database"),
+        ],
+        editors=[
+            base_editor,
+            awb.BasicCellEditor(title="Cell Editor"),
         ],
+        viewer=awb.viewers.StructureDataViewer(),
+        node_class="StructureData",
     )
 
     assert structure_manager_widget.structure is None
+    structure_manager_widget.input_structure = stored_structure
+
+    # Set action vector perpendicular to screen.
+    base_editor.def_perpendicular_to_screen()
```

Comment:
I assume this changes the axis, would be better to add an assert for the effect.

Reply:
I am not sure how to check that properly. It depends on the starting orientation of the viewer.